### PR TITLE
[UPDATE] sonar action configuration

### PR DESCRIPTION
**Pull request by sethkfman**
**Development & PR Process**
1. Follow MetaMask [Mobile Coding Standards](https://docs.google.com/document/d/1VJLwTRsUw_5EDq_o8d6sSbXUAYENLurkRitYO45eY5o/edit?usp=sharing)
2. Add `release-xx` label to identify the PR slated for a upcoming release (will be used in release discussion)
3. Add `needs-dev-review` label when work is completed
4. Add `needs-qa` label when dev review is completed
5. Add `QA Passed` label when QA has signed off

**Description**

The sonar build action needed configuration updates after it ran
- Updated checkout action to remove node version warning in action
- Added configs for the project analysis
```
             -Dsonar.javascript.lcov.reportPaths=tests/coverage/lcov.info
             -Dsonar.exclusions=android/,ios/,storybook/,sources/,scripts/,e2e/,wdio/
             -Dsonar.tests=app/
             -Dsonar.test.inclusions=**/*.test.*
```
**Screenshots/Recordings**

NA

**Issue**

NA

**Checklist**

* [NA] There is a related GitHub issue
* [NA] Tests are included if applicable
* [NA] Any added code is fully documented

### CODE REVIEW

**Review by Cal-L — APPROVED** [review]
Lgtm

File: .github/workflows/sonar.yml [hunk 1]
```diff
@@ -8,10 +8,15 @@ jobs:
...
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
         with:
           fetch-depth: 0
       - uses: sonarsource/sonarqube-scan-action@427bad70165a74f9d7133d48ea31b482a53ee9fa
+        with:
+          args: >
+            -Dsonar.javascript.lcov.reportPaths=tests/coverage/lcov.info
+            -Dsonar.tests=tests/
```
Comment (Gudahtt) [inline .github/workflows/sonar.yml:18]: I was able to find this configuration referenced here: https://docs.sonarqube.org/latest/analyzing-source-code/analysis-parameters/

It looks like this is for pointing Sonar to where test source files are, but I don't see any tests in the `tests/` directory. Mobile seems to be using that directory just for coverage reports.

Comment (sethkfman) [reply .github/workflows/sonar.yml:18]: @Gudahtt  Correct, I opted to remove the parameter. 

Right now our tests are spread throughout the project and mixed in with our source files. We will have to see how this impacts the code analysis.

Comment (sethkfman) [reply .github/workflows/sonar.yml]: @Gudahtt Set the root to the app folder which is where all developer level tests are located. Also, used a more general regex to cover all js, ts, jsx & tsx test files. 
```
            -Dsonar.tests=app/
            -Dsonar.test.inclusions=**/*.test.*
```

Comment (Gudahtt) [reply .github/workflows/sonar.yml]: Looks good! Hopefully that works

**Review by Gudahtt — APPROVED** [review]

**Pull request merged**